sdk/core/core-http/lib/util/xml.ts

```diff
@@ -34,4 +33,171 @@ export function parseXML(str: string): Promise<any> {
+  xml2jsSettings.normalize = true;
+  return xml2jsSettings;
+}
+
+/**
+ * @ignore
+ * @param str Helper utility to clean up unintended characters that get appended by OS.
+ */
+function _removeBOM(str: any) {
+  if (str.charCodeAt(0) === 0xfeff || str.charCodeAt(0) === 0xffef) {
+    str = str.substring(1);
+  }
+
+  return str;
+}
+
+/**
+ *
+ * @ignore
+ * Writes a single property for an entry or complex type.
+ *
+ * @param {object} parentElement         Parent DOM element under which the property should be added.
+ * @param {string} name                  Property name.
+ * @param {object} value                 Property value.
+ * @return {object} The current DOM element.
+ *
+ * {Deprecated}
+ */
+function _writeElementValue(parentElement: any, name: any, value: any): any {
+  let ignored = false;
+  const propertyTagName = name;
+
+  if (!contentIsUndefinedOrEmpty(value) && isObject(value) && !isDate(value)) {
+    if (Array.isArray(value) && value.length > 0) {
+      // Example:
+      // JSON: element: [ { property1: 'hi there' }, { property2: 'hello there' } ]
+      // XML: <element><property1>hi there</property1></element><element><property2>hello there</property2></element>
+
+      Object.keys(value).forEach(function(i: any) {
+        parentElement = _writeElementValue(parentElement, name, value[i]);
+      });
+
+      // For an array no element was actually added at this level, so skip uping level.
+      ignored = true;
+    } else if (
+      value[Constants.XML_METADATA_MARKER] !== undefined &&
+      value[Constants.XML_VALUE_MARKER] !== undefined
+    ) {
+      // Example:
+      // JSON: element: { '$': { 'm:type' = 'Edm.String' }, '_': 'hi there' }
+      // XML: <element m:type="Edm.String">hi there</element>
+
+      parentElement = parentElement.ele(propertyTagName);
+      if (!contentIsUndefinedOrEmpty(value[Constants.XML_VALUE_MARKER])) {
+        if (new String(value[Constants.XML_VALUE_MARKER]).startsWith("<![CDATA[")) {
+          parentElement = parentElement.raw(value[Constants.XML_VALUE_MARKER]);
+        } else {
+          parentElement = parentElement.txt(value[Constants.XML_VALUE_MARKER]);
+        }
+      }
+    } else {
+      // Example:
+      // JSON: element: { '$': { 'metadata' = 'value' }, 'property1': 'hi there', 'property2': 'hello there' }
+      // XML: <element metadata="value"><property1>hi there</property1><property2>hello there</property2></element>
+
+      parentElement = parentElement.ele(propertyTagName);
+      Object.keys(value).forEach((propertyName: string) => {
+        if (propertyName !== Constants.XML_METADATA_MARKER && value.hasOwnProperty(propertyName)) {
+          parentElement = _writeElementValue(parentElement, propertyName, value[propertyName]);
+        }
+      });
+    }
+  } else {
+    parentElement = parentElement.ele(propertyTagName);
+    if (!contentIsUndefinedOrEmpty(value)) {
+      if (new String(value.toString().trim()).startsWith("<![CDATA[")) {
```

Comment:
Based on usage, it can be an object with key value properties as well.

Reply:
> Based on usage, it can be an object with key value properties as well.

Isn't that why you are calling `value.toString()`? That should already convert the object to a string.

Reply:
Oh, you meant the wrapping - yeah, it must have been added to get a 'String' object originally. Will update.

Reply:
Updated PR with this and few other cleanup items